clients/cobol-lsp-vscode-extension/src/test/suite/processorgroups.spec.test.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as assert from "assert";
+import * as helper from "./testHelper";
+import * as vscode from "vscode";
+import * as path from "path";
+
+suite("Integration Test Suite: Processor Groups", function () {
+  suiteSetup(async function () {
+    this.timeout(helper.TEST_TIMEOUT);
+    helper.updateConfig("basic.json");
+    await helper.activate();
+  });
+
+  this.afterEach(async () => await helper.closeAllEditors()).timeout(
+    helper.TEST_TIMEOUT,
+  );
+
+  // cobol-dialect-api (development) npm install && npm run compile
+  // daco-dialect-support (development) npm install && npm run package
+  // idms-dialect-support (development) npm install && npm run package
+  // put server.jar, dialect-idms.jar and dialect-daco.jar into their server/jar folder
+
+  test("TC355920: IDMS - processor group", async () => {
+    const extSrcPath = path.join("cobol-idms", "DACRFI11N.cbl");
+    await helper.showDocument(extSrcPath);
+    const editor = helper.get_editor(extSrcPath);
+    const waitResult = await helper.waitFor(
+      () => vscode.languages.getDiagnostics(editor.document.uri).length == 1,
+      5000
+    );
+    const diagnostics = vscode.languages.getDiagnostics(editor.document.uri);
+    assert.strictEqual(waitResult, false);
```

Comment:
This means that the waitFor timed out, right?

Reply:
right

Reply:
So, are you planning on fixing that?

Reply:
that is done on purpose 
there is sometime for the processing to run and them the confirmation that no error was found

Reply:
The waitFor callback tests `() => vscode.languages.getDiagnostics(editor.document.uri).length == 1`. If 2 diagnostics arrive the test will pass as well.

Introduce a source of ONE diagnostic into e.g. SUBSCHEMA-NAMES.cpy and match it in the test.

Reply:
I am expecting the diagnostic to be zero
I don't see how introducing a one diagnostic would help, can you elaborate more on that?

Reply:
Because you cannot test for something to NOT HAPPEN. So write a test that actually produces something that can be tested.

Reply:
this test produces no errors which is what is being tested

Reply:
Except that's not what is being tested here. You are actually testing if exactly one diagnostic is not produced within 5 seconds.

Reply:
that is just to give some time to finish the processing
the actual test happens here
`assert.strictEqual(diagnostics.length, 0);`

Reply:
How can you tell from this test that the server did not crash or is in an infinite loop?